main.js

```diff
@@ -45,6 +48,9 @@ module.exports = class HotReload extends Plugin {
         for (const dir of Object.keys(this.pluginNames)) {
             for (const file of ["manifest.json", "main.js", "styles.css", ".hotreload"]) {
                 const path = `${base}/${dir}/${file}`;
+                if (!fs.existsSync(path)) {
```

Comment:
this block is unnecessary -- Obsidian's adapter.stat traps ENOENT errors and simply returns null.